ts/Extensions/Annotations/Popup.ts

```diff
@@ -830,17 +861,62 @@ H.Popup.prototype = {
                 buttonParentDiv as any,
                 lang.saveButton || 'save',
                 'edit',
-                callback,
-                buttonParentDiv as any
+                buttonParentDiv as any,
+                callback
             );
             this.addButton(
                 buttonParentDiv as any,
                 lang.removeButton || 'remove',
                 'remove',
-                callback,
-                buttonParentDiv as any
+                buttonParentDiv as any,
+                callback
             );
         },
+        /**
+         * Sort the series in the object.
+         * @private
+         */
+        sortSeries: function (series: Highcharts.FilteredSeries): Highcharts.FilteredSeries {
+            let sorted: Highcharts.FilteredSeries = {},
+                array = [],
+                key;
+
+            for (key in series) {
+                if (Object.prototype.hasOwnProperty.call(series, key)) {
+                    array.push(key);
+                }
+            }
+
+            array.sort();
+
+            for (key = 0; key < array.length; key++) {
+                sorted[array[key]] = series[array[key]];
+            }
+            return sorted;
+        },
+        /**
+         * Create an array of filtered and sorted series based on the filter.
+         * @private
+         */
+        filterSeries: function (series: Highcharts.FilteredSeries, filter?: string): Highcharts.FilteredSeries {
+            let filteredSeries = {} as Highcharts.FilteredSeries;
```

Comment:
Just a discussion point. It seems that to create a list of indicators we do this:
```
A) Get the list for ADD/EDIT:
  a.1) If it's EDIT -> use array from `chart.series`
  a.2) If it's ADD -> use the object `plotOptions`
    a.2.1) Filter out series which are not indicators
    a.2.2) Filter out series which don't match the name from input 
    a.2.3) Sort series:
      a.2.2.1) Map object to array
      a.2.2.1) Sort array
      a.2.2.1) Map array to object
```
    
 How about:
 - we merge a.2.1) and a.2.2)?
 - we do the mapping from object to array in the first filter, and later use only an array of series (objects)? That would simplify TS (no Highcharts.FitleredSeries interface) and reduce the number of unnecessary loops/iterations over an object.